### PR TITLE
workflows: bump license_finder to v7.0.1

### PR DESCRIPTION
**Pull request by bobheadxi**
Improves support for yarn v2, which we use now: https://github.com/pivotal/LicenseFinder/releases/tag/v7.0.1

I believe we're missing yarn dependencies from the third party license right now

## Test plan

https://github.com/sourcegraph/sourcegraph/actions/runs/3376178761

### CODE REVIEW

**Review by burmudar — APPROVED** [review]

File: third-party-licenses/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+*   @sourcegraph/dev-experience
```
Comment (bobheadxi) [inline third-party-licenses/CODEOWNERS:1]: @sourcegraph/dev-experience the generated PRs from this workflow requires stamp approvals + merge

Comment (burmudar) [reply third-party-licenses/CODEOWNERS:1]: right - I'll take a look at them. Thanks for flagging it @bobheadxi

**Comment by burmudar** [comment]
@bobheadxi successful run https://github.com/sourcegraph/sourcegraph/actions/runs/3378717520 !

**Comment by burmudar** [comment]
PR that got generated https://github.com/sourcegraph/sourcegraph/pull/43814

**Comment by burmudar** [comment]
> @burmudar if it passes that's a good start, but looks like we still don't have yarn dependencies listed in there :( At some point in the past we definitely did

A damn. Gotcha. I'll look at that

**Comment by bobheadxi** [comment]
If it's a problem with license_finder happy to have this merged first, but I think we need to figure a way forward here eventually 😬

**Comment by burmudar** [comment]
It might be yarn. We're using a plugin now but I'll have to see how license_finder parses the output. Because before it couldn't execute the license command and now it can, but now license finder isn't doing something with the output.

**Comment by burmudar** [comment]
License Finders support for yarn 2 / 3 is still a bit shaky. Found this discussion on it https://groups.google.com/g/license-finder/c/yaJeO1okF4c. A fixed has been merged in Sept but there has not been a new release. I can try building the package from the branch but that won't be sustainable.

We might have to investigate other tools. I found https://scancode-toolkit.readthedocs.io/en/stable/getting-started/home.html but will have to check if it handles yarn better than License Finder

**Pull request merged**